### PR TITLE
【translate】mds/schedule

### PR DESCRIPTION
**Pull request by SeanHai**
### What problem does this PR solve?

Issue Number: close #xxx 

Problem Summary:

### What is changed and how it works?

What's Changed:

How it Works:

Side effects(Breaking backward compatibility? Performance regression?): 

### Check List

- [ ] Relevant documentation/comments is changed or added
- [ ] I acknowledge that all my contributions will be made under the project's license

### CODE REVIEW

**Review by cw123 — CHANGES_REQUESTED** [review]

File: src/mds/schedule/coordinator.cpp [hunk 1]
```diff
@@ -201,8 +207,8 @@ int Coordinator::QueryChunkServerRecoverStatus(
         infos.emplace_back(info);
     }
 
-    // 查询每个chunkserver是否正在恢复
-    // 正在恢复：offline且有恢复任务；其他均为未在恢复
+    // Iterate whether each chunkserver is recovering
```
[the remark this answers is not in the document]
Comment (SeanHai) [reply src/mds/schedule/coordinator.cpp:210]: > 查询拼错了

done

File: src/mds/schedule/coordinator.h [hunk 2]
```diff
@@ -79,93 +77,97 @@ class Coordinator {
...
     */
     std::shared_ptr<OperatorController> GetOpController();
 
  private:
     /**
-     * @brief  SetScheduleRunning，如果设置为false,则停止所有的scheduelr
+     * @brief SetScheduleRunning Stop all the schedulers if set to false
      *
-     * @param[in] flag 为false，所有sheduler会停止
+     * @param[in] flag
```
[the remark this answers is not in the document]
Comment (SeanHai) [reply src/mds/schedule/coordinator.h:132]: > 没有翻译

上面brief中已经描述了。
     * @brief SetScheduleRunning Stop all the schedulers if set to false
     *
     * @param[in] flag

File: src/mds/schedule/copySetScheduler.cpp [hunk 3]
```diff
@@ -54,23 +54,33 @@ int CopySetScheduler::DoCopySetSchedule(PoolIdType lid) {
...
-    // - copyset成员发生变化后，成员上的scatter-with的变化要满足一定的要求
-    //   * 成员上的[scatter-with < 最小值]，变动不能使得scatter-with减小
-    //   * 成员上的[scatter-with > 最大值]，变动不能使scatter-with更大 //NOLINT
-    //   * 成员上的[ 最小值 <= scatter-with <= 最大值]，可以随意变动
-    // 最大值和最小值的确定：
-    // - 最小值由参数配置
-    // - 最大值为最小值上浮一定百分比， 这种确定方式使得极差在一定范围之内
+    /**
+     * 3. Set migration condition
+     *    condition: range and average volume fall into a certain percentage
```
[the remark this answers is not in the document]
Comment (SeanHai) [reply src/mds/schedule/copySetScheduler.cpp:65]: > 条件：极差超过均值的百分比范围，开始迁移

done

File: src/mds/schedule/leaderScheduler.cpp [hunk 4]
```diff
@@ -67,7 +67,8 @@ int LeaderScheduler::DoLeaderSchedule(PoolIdType lid) {
         }
 
         if (minLeaderCount == -1 || csInfo.leaderCount < minLeaderCount) {
-            // 因为只有minLeaderCount的才会作为目标节点，这里只需要判断目标节点是否刚启动
+            // the chunkserver with the start up time within the cooling time
+            // can not be the leader
```
[the remark this answers is not in the document]
Comment (SeanHai) [reply src/mds/schedule/leaderScheduler.cpp:71]: > 翻译意思好像不对

done

File: src/mds/schedule/recoverScheduler.cpp [hunk 5]
```diff
@@ -70,12 +70,12 @@ int RecoverScheduler::Schedule() {
             }
         }
 
-        // 如果所有的copyset副本都是online状态，不做处理
+        // do nothing if all replicas are offline
```
[the remark this answers is not in the document]
Comment (SeanHai) [reply src/mds/schedule/recoverScheduler.cpp:73]: > 意思反了

done

File: src/mds/schedule/recoverScheduler.cpp [hunk 6]
```diff
@@ -104,35 +104,36 @@ int RecoverScheduler::Schedule() {
             continue;
         }
 
-        // 修复其中一个挂掉的副本
+        // recover onr of the offline replica
```
[the remark this answers is not in the document]
Comment (SeanHai) [reply src/mds/schedule/recoverScheduler.cpp:107]: > one

done

File: src/mds/schedule/scheduler.h [hunk 7]
```diff
@@ -430,62 +457,63 @@ class RapidLeaderScheduler : public Scheduler {
         : Scheduler(opt, topo, opController), lpoolId_(lpid) {}
 
     /**
-     * @brief 以逻辑池为力度做leader均衡
-     * @return 返回值是一个:
-     *         kScheduleErrCodeSuccess 成功生成了一批transferleader的operator
+     * @brief Execute leader balancing in a granularity of logical pool
+     * @return kScheduleErrCodeSuccess
```
[the remark this answers is not in the document]
Comment (SeanHai) [reply src/mds/schedule/scheduler.h:461]: > return 的意思

done

File: src/mds/schedule/scheduler_helper.cpp [hunk 8]
```diff
@@ -30,60 +30,61 @@
...
         } else {
-            // target, 变更之后没有使得scatter-with至少减少1，不符合条件
+            // the scatter-width decrease is less than 1, invalid
             if (newValue - oldValue >= 0) {
                 return false;
             }
         }
     }
-    // 变更之后的scatter-with在[min, max]之间
+    // the newValue falls inside [min, max]
```
[the remark this answers is not in the document]
Comment (SeanHai) [reply src/mds/schedule/scheduler_helper.cpp:87]: > 这里的注释应该有问题，这个时候，newValue可能还有在[min, max]之外

delete

File: src/mds/schedule/scheduler_helper.cpp [hunk 9]
```diff
@@ -30,60 +30,61 @@
...
+ *   ② newValue<minScatterWidth, but the scatter-width didn't drop
+ *     (newValue-oldValue>=0)
+ *   ③ newValue>maxScatterWidth, but the scatter-width didn't increase
+ *     (newValue-oldValue<=0)
+ * 2. for D
+ *   ① minScatterWidth<=newValue<=maxScatterWidth
+ *   ② newValue<minScatterWidth, but the scatter-width increased at least 1
+ *     (newValue-oldValue>=1)
+ *   ③ newValue>maxScatterWidth, but the scatter-width decreased at least 1
+ *     (newValue-oldValue<=-1)
```
[the remark this answers is not in the document]
Comment (SeanHai) [reply src/mds/schedule/scheduler_helper.cpp:50]: > 加上if条件

done

File: src/mds/schedule/scheduler_helper.cpp [hunk 10]
```diff
@@ -30,60 +30,61 @@
...
+ * for operation copyset(ABC),-C,+D, C is the source and D is the target.
+ * this operation will change the scatter-width of {ABCD}, and for old
+ * scatter-width oldValue and new scatterwidth newValue, the change should
+ * satisfy one of the conditions below:
+ * 1. for A,B,C
+ *   ① minScatterWidth<=newValue<=maxScatterWidth
+ *   ② newValue<minScatterWidth, but the scatter-width didn't drop
+ *     (newValue-oldValue>=0)
+ *   ③ newValue>maxScatterWidth, but the scatter-width didn't increase
+ *     (newValue-oldValue<=0)
```
[the remark this answers is not in the document]
Comment (SeanHai) [reply src/mds/schedule/scheduler_helper.cpp:44]: > 加上if条件

done

**Review by cw123 — APPROVED** [review]

**Review by cw123 — APPROVED** [review]

**Review by ilixiaocui — APPROVED** [review]

**Pull request merged**